File: md-shamim-ahmad_LeetCode-Problem-Solution/LeetCode Problems/1501 to 1600/1544. Make The String Great/1544. Make The String Great.py
class Solution:
    def makeGood(self, s: str) -> str:
        s = list(s)
        i = 0
        while i < len(s) - 1:
            if s[i].lower() == s[i+1].lower() and s[i].isupper() == s[i + 1].islower():
                s.pop(i)
                s.pop(i)
                i -= 1 if i > 0 else 0
                continue
            i += 1
        return "".join(s);

File: md-shamim-ahmad_LeetCode-Problem-Solution/LeetCode Problems/1501 to 1600/1544. Make The String Great/1544. Make The String Great.cpp
#include <bits/stdc++.h>
#include <vector>
#include <iostream>
#include <string>
#include <map>
#include <set>
#include <unordered_map>
#include <algorithm>
using namespace std;

class Solution {
public:
    string makeGood(string s) {
        stack<char> stc;
        for (int i = 0; i < s.size(); i++) {
            if (stc.empty()) {
                stc.push(s[i]);
            } else {
                if ((stc.top() == tolower(s[i]) || stc.top() == toupper(s[i])) && stc.top() != s[i]) {
                    stc.pop();
                } else {
                    stc.push(s[i]);
                }
            }
        }
        s = "";
        while (!stc.empty()) {
            s += stc.top();
            stc.pop();
        }
        reverse(s.begin(), s.end());
        return s;
    }
};